File: solutions/lua/scrabble-score/1/scrabble-score.lua
local letter_scores = {
    A = 1, E = 1, I = 1, O = 1, U = 1, L = 1, N = 1, R = 1, S = 1, T = 1, D = 2, G = 2,
    B = 3, C = 3, M = 3, P = 3,
    F = 4, H = 4, V = 4, W = 4, Y = 4,
    K = 5,
    J = 8, X = 8,
    Q = 10, Z = 10,
}

local function score(word)
    local _score = 0

    if word == nil then
        return _score
    end

    for i = 1, #word do
        local letter = string.upper(word:sub(i,i))
        _score = _score + letter_scores[letter]
    end

    return _score
end

return { score = score }

File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble

import "unicode"

func Score(word string) int {
	var score int = 0

	letters := []rune(word)

	for i := 0; i < len(letters); i++ {
		letter := letters[i]
		score += LetterPoints(letter)
	}

	return score
}

func LetterPoints(letter rune) int {
	switch unicode.ToUpper(letter) {
	case 'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T':
		return 1
	case 'D', 'G':
		return 2
	case 'B', 'C', 'M', 'P':
		return 3
	case 'F', 'H', 'V', 'W', 'Y':
		return 4
	case 'K':
		return 5
	case 'J', 'X':
		return 8
	case 'Q', 'Z':
		return 10
	default:
		return 0
	}
}
